### PR TITLE
chore: Improve the CI on PRs to be more efficient

### PR DESCRIPTION
**Pull request by JorTurFer**
### Checklist
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

### CODE REVIEW

File: .github/workflows/ci-external-scaler-azure-cosmos-db.yml [hunk 1]
```diff
@@ -35,26 +39,14 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        kubernetesVersion: [v1.24, v1.23, v1.22, v1.21, v1.20, v1.19, v1.18, v1.17]
+        kubernetesVersion: [v1.24, v1.19]
```
Comment (zroubalik) [inline .github/workflows/ci-external-scaler-azure-cosmos-db.yml:42]: Should we test here the latest version only?

@tomkerkhove

Comment (tomkerkhove) [reply .github/workflows/ci-external-scaler-azure-cosmos-db.yml:42]: Let's follow our support matrix for KEDA and not x Kubernets versions. We should only test that, the rest is on their own risk imo

Comment (JorTurFer) [reply .github/workflows/ci-external-scaler-azure-cosmos-db.yml:42]: So, can I merge this PR?

Comment (zroubalik) [reply .github/workflows/ci-external-scaler-azure-cosmos-db.yml:42]: Shouldn't we replace 1.24 with the latest one?

Comment (JorTurFer) [reply .github/workflows/ci-external-scaler-azure-cosmos-db.yml:42]: I prefer to not touch their versions. I mean, if their scaler doesn't work on 1.28? Now I have just pruned up their versions and that's all, nothing more has changed

**Review by zroubalik — APPROVED** [review]

**Pull request merged**